src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyLinkedList.Node<T> first;
+    private MyLinkedList.Node<T> last;
+
+    MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+        } else {
+            Node node = new Node<>(last, value, null);
+            last.setNext(node);
+            last = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        validateAddIndex(index);
+        Node node = first;
+        Node newNode = new Node<>(value);
+        if (node == null) {
+            first = newNode;
+            last = newNode;
+            size++;
+            return;
+        }
+        for (int i = 0; i <= size; i++) {
+            if (index == i) {
+                if (node == null) {
+                    last.setNext(newNode);
+                    newNode.setPrev(last);
+                    last = newNode;
+                } else if (i == 0) {
+                    node.setPrev(newNode);
+                    newNode.setNext(node);
+                    first = newNode;
+                } else {
+                    newNode.setPrev(node.getPrev());
+                    node.getPrev().setNext(newNode);
+                    node.setPrev(newNode);
+                    newNode.setNext(node);
+                }
+                size++;
+                return;
+            }
+            node = node.getNext();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node node = last;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        validateIndex(index);
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                return (T) node.getItem();
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        validateIndex(index);
+        final T replacedElement;
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                replacedElement = (T) node.getItem();
+                node.setItem(value);
+                return replacedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        validateIndex(index);
+        final T removedElement;
+        Node node = first;
```

Comment:
Look through your code and parametrize all your Node (Node -> Node<T>)
```suggestion
        Node<T> node = getNodeByIndex(index);
```

Reply:
done